tests/sentry/web/frontend/test_accounts.py

```diff
@@ -148,6 +148,31 @@ def test_relocate_recovery_post(self):
         assert resp.status_code == 302
         assert resp[header_name] == "strict-origin-when-cross-origin"
 
+    def test_relocate_recovery_unchecked_tos(self):
+        user = self.create_user()
+
+        resp = self.client.post(self.path, {"user": user.email})
+        assert resp.status_code == 200
+
+        lost_password = LostPasswordHash.objects.get(user=user)
+        user.is_unclaimed = True
+        user.save()
+        new_username = "test_username"
+
+        resp = self.client.post(
+            self.relocation_recover_path(lost_password.user_id, lost_password.hash),
+            {"username": new_username, "password": "test_password", "tos_check": False},
+        )
+
+        header_name = "Referrer-Policy"
+
+        user.refresh_from_db()
+        assert resp.has_header(header_name)
+        assert resp.templates[0].name == ("sentry/account/relocate/confirm.html")
+        assert user.is_unclaimed
+        assert resp.status_code == 200
```

Comment:
Took a swing at it, but returning a non-2XX status code when the form data is invalid will not properly report validation errors in the form for password validation, it will redirect to a screen that says "Your form is invalid" or which is not the best user experience. I think for our purposes, this is ok for the following reasons:
1. The form forces people to check the box before actually sending the POST request
2. This test checks that if a POST request actually makes it and for some reason the `tos_check` is false, it will not do anything

Reply:
I added a more detailed message `You must agree to the Terms of Service and Privacy Policy before proceeding.` if people do not check the box and tested for that.

Reply:
👍